exercises/practice/leap/.approaches/operators/content.md

```diff
@@ -0,0 +1,98 @@
+# Boolean operators
+
+```elixir
+defmodule Year do
+  @spec leap_year?(non_neg_integer) :: boolean
+  def leap_year?(year) do
+    rem(year, 4) == 0 and not rem(year, 100) == 0 or rem(year, 400) == 0
+  end
+end
+```
+
+## Short-circuiting
+
+At the core of this approach, three checks are returning three boolean values. 
+We can use [Boolean logic](https://en.wikipedia.org/wiki/Boolean_algebra) to combine the results.
+
+When using this approach, it is essential to consider short-circuiting of boolean operators. 
```

Comment:
I'd say short-circuiting does not matter at all in this exercise. Short-circuiting is important to consider when we want to avoid or ensure some side effects, for example:

```
iex()> 40 / x
** (ArithmeticError) bad argument in arithmetic expression: 40 / 0
    :erlang./(40, 0)
    iex:3: (file)

iex()> x != 0 && 40 / x
false
```

In this exercise, all inputs are valid, all the checks on either side of the boolean operator are pure functions with no side effects.

Reply:
Here are the reasons why I think it is still relevant, even if all inputs are valid, all the checks are pure functions. 

1. In the approach I'm showing different variations of the code. The option with the custom `divides?/2`  function is not necessarily guaranteed to be pure. 
2. The approach at the end I'm showing an option to capture the results in variables. Perhaps I should make it more explicit, but in this case, all checks are performed because the short-circuiting applies only to checking the value of the variables. 
3. In every exercise instructions, at the bottom, there instructions on how to use `IO.inspect/1` for debugging your solutions. It is possible, that somebody less experienced will be surprised that inspect added to a custom function, or perhaps directly to one of the checks. Consider this code

```elixir
rem(year, 4) == 0 and not IO.inspect(rem(year, 100) == 0) or rem(year, 400) == 0
```
When `rem(year, 4) != 0` the `IO.inspect/1` will not show anything because it will not be evaluated.

Reply:
ok, I'm convinced 🙂